private/buf/cmd/buf/command/build/build.go

```diff
@@ -139,6 +140,21 @@ func run(
 	if err != nil {
 		return err
 	}
+	imageRef, err := buffetch.NewImageRefParser(container.Logger()).GetImageRef(ctx, flags.Output)
+	if err != nil {
+		return fmt.Errorf("--%s: %v", outputFlagName, err)
+	}
+	var buildOptions []bufimagebuild.BuildOption
+	if flags.ExcludeSourceInfo {
+		buildOptions = append(buildOptions, bufimagebuild.WithExcludeSourceCodeInfo())
+	} else if !flags.AsFileDescriptorSet {
+		// if not a descriptor set, not necessary to try to match protoc, so include extra comments
+		buildOptions = append(buildOptions, bufimagebuild.WithExtraSourceCodeInfo())
+	}
+	if flags.AsFileDescriptorSet && imageRef.ImageEncoding() == buffetch.ImageEncodingBin {
+		// if generating binary format, emit the same bytes as protoc
```

Comment:
Effectively same comment - I don't think we need to lock ourselves long-term into emitting the exact same bytes as protoc. We fulfill the contract that protoc has by producing functionally-equivalent FileDescriptorSets, our definition of what these are is just as good.

Reply:
This was basically the fix for https://github.com/jhump/protoreflect/issues/214

If you don't think we should do that, it's easy to remove. But I thought there was value in it.

Reply:
Let's discuss tomorrow/in person this week